src/tools/ethBalanceCheck/eth-balance-check.ts

```diff
@@ -1,20 +1,52 @@
 import Web3 from "web3";
+import { ChainConfig } from "delphinus-deployment/src/types";
 
-export async function checkEthBalance(provider: string, address: string, warningAmount: string) {
-  let ethBalanceWarning = false;
+export async function checkDeployerAccountBalance(config: ChainConfig, warningAmount: string){
+  if (config.privateKey === "") {
+    return checkEthBalance(config.wsSource, config.privateKey, config.deviceId, warningAmount);
+  } else {
+    return checkEthBalance(config.rpcSource, config.privateKey, config.deviceId, warningAmount);
+  }
+}
+
+export async function checkEthBalance(providerSource: string, privateKey: string, ChainId: string, warningAmount: string) {
+  let balanceWarning = false;
   let balance;
-  const web3 = new Web3(new Web3.providers.HttpProvider(provider));
+  const web3 = getWeb3FromSource(providerSource);
+  const address = web3.eth.accounts.privateKeyToAccount(privateKey).address;
+  const currencySymbol = await getNativeCurrencySymbolByChainId(ChainId);
   await web3.eth.getBalance(address, function(err, result) {  
     if (err) {
       console.log(err)
     } else {
       balance = Web3.utils.fromWei(result, "ether");
-      if (parseInt(result, 10) < parseInt(Web3.utils.toWei(warningAmount, "ether"), 10)){
-        ethBalanceWarning = true;
-        console.log("Warning: Your ETH Balance is less than " + warningAmount + " ETH");
-        console.log("Your current ETH balance: " + balance + " ETH");
+      if (parseInt(balance, 10) < parseInt(warningAmount, 10)){
+        balanceWarning = true;
+        console.log("Warning: Deployer's balance is less than WarningAmount(" + warningAmount + " " + currencySymbol + ")");
+        console.log("Deployer's balance: " + balance + " " + currencySymbol);
       }
     }
   });
-  return [ethBalanceWarning, balance]
+  return [balanceWarning, balance, currencySymbol]
+}
+
+export function getWeb3FromSource(provider: string) {
+  const HttpProvider = "https";
+  if(provider.includes(HttpProvider)){
+    return new Web3(new Web3.providers.HttpProvider(provider));
+  }else {
+    return new Web3(new Web3.providers.WebsocketProvider(provider));
+  }
+}
+
+export async function getNativeCurrencySymbolByChainId(chainId: string){
+  const chainInfo = await getChainInfoByChainID(chainId);
+  return chainInfo.nativeCurrency.symbol;
+}
+
+export async function getChainInfoByChainID(chainId:string) {
+  const fetch = require("node-fetch");
```

Comment:
node-fetch didn't included in the package.json.
You can use it in lerna env is just because deployment repo and ts-sdk repo is using it.
Thus it need to be added to the monitor package.json with:
"node-fetch": "^2.6.6"

Reply:
added